xclim/indicators/land/_snow.py

```diff
@@ -32,36 +38,105 @@ class SnowWithIndexing(ResamplingIndicatorWithIndexing):
 
 
 snow_cover_duration = SnowWithIndexing(
-    title="Snow cover duration",
+    title="Snow depth cover duration",
     identifier="snow_cover_duration",
     units="days",
     long_name="Number of days with snow depth at or above threshold",
     description="The {freq} number of days with snow depth greater than or equal to {thresh}.",
     abstract="Number of days when the snow depth is greater than or equal to a given threshold.",
     compute=xci.snow_cover_duration,
+    _version_deprecated="0.41.0",
+)
+
+snow_depth_cover_duration = SnowWithIndexing(
+    title="Snow depth cover duration",
```

Comment:
```suggestion
```
`title` and `abstract`  are unneeded here if `xclim` can gather them from the docstring.

Reply:
Hi guys,
in function `snow_depth_from_amount` and `snow_amount_from_depth` in `xclim/indices/_conversion.py` the snow density `snr` is an input parameter. I suggets to use the default value from [Sturm et. al 2010](https://journals.ametsoc.org/view/journals/hydr/11/6/2010jhm1202_1.xml). This is 312 kg*m**3.
What do you think about it?

Cheers, Ludwig

Reply:
@ludwiglierhammer

Interesting! I was looking for a source to grab a reasonable default value from, but never found anything. I'll give that article a look, and maybe we can run with that number. Presently, it asks for the density as a DataArray, which is fine as ECMWF and ESGF products tend to have this variable available. Will let you know!

Reply:
I think it is better here to be fully explicit because this value, even though it represents well the average, is still only an approximation of something that can greatly vary! 
To illustrate, here is the distribution of 400k snow density measurements all over Québec:
![image](https://user-images.githubusercontent.com/20629530/219067196-1d8c83ff-6369-4f8e-9046-9ad3827db023.png)

However, passing a single value might be sometimes useful, but not possible right now as a DataArray is expected and passing a string is understood as a variable name on the input dataset... It might be nice to find a way, at least for these methods, where one can easily pass a constant value instead.

The simplest I could find is :
```python3
xc.land.snow_depth_from_amount(ds.snw, xr.DataArray(312, attrs={'units': 'kg m-3'}))
```

Reply:
This is a very good objection. However, with many projects like `CMIP` or `CORDEX` the snow density varibale `snr` is not requested. For instance, one can not use `snr`from `ECMWF` with climate scenarios  (`RCP`'s). I think a snow density default value can be very helpful in this case.

What do you think about something like this: 

```
@declare_units(snd="[length]", snr="[mass]/[volume]")
def snow_amount_from_depth(
    snd: xr.DataArray,
    snr: xr.DataArray = None,
    const: Quantified = "312 kg m-3",
) -> xr.DataArray:
    """Snow amount from snow depth and density.
    Parameters
    ----------
    snd : xr.DataArray
        Snow depth [m].
    snr : xr.DataArray, optional
        Snow density [kg/m^3].
   const: Quantified
        Constant snow density [kg/m^3]
    Returns
    -------
    xr.DataArray, [kg m-2]
        Surface snow amount
    """
    if snr is None:
       const = convert_units_to(const, "kg m-3")
       snr = xr.DataArray(const, attrs={"units": "kg m-3"})
    snd = convert_units_to(snd, "m")
    snr = convert_units_to(snr, "kg m-3")

    snw = snd * snr

    snw.attrs["units"] = "kg m-2"
    return snw
```

Reply:
If someone uses `const` maybe we could print a warning message.

Reply:
I agree with this implementation. With some addition to the docstring to make clear that `const` is not used if `snr` is given. And the unit logic can be simplified even more because `convert_units_to` will return a "DataArray-multipicable" object anyway.
```python3
if snr is None:
    snr = const
snd = convert_units_to(snd, "m")
snr = convert_units_to(snr, "kg m-3")

snw = snd * snr
```

Reply:
I like this approach as well!

@ludwiglierhammer If you'd like to push those changes into the code base, that would be appreciated! 

If you'd also be able to add the reference to https://journals.ametsoc.org/view/journals/hydr/11/6/2010jhm1202_1.xml into the BibTeX library (https://github.com/Ouranosinc/xclim/blob/master/docs/references.bib), you could then add that citation into the docstring with `:cite:p:`, `:cite:t:`, or `:cite:cts:`.

Reply:
Should I create a pull_request into `add_snow_water_cover_indices` or should I wait until you merged this pull_request into your `master` and then create a new pull_request? What is handier for your workflow?

Reply:
@ludwiglierhammer

I see no issue with you opening a PR into this branch. These changes aren't blocking to this PR at hand, and when this PR is eventually merged, the snow density PR will rebase onto `master`